lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(obj)
+    @name = obj[:name]
+    @address = obj[:address]
+    @phone = obj[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @registered_vehicles << vehicle
+
+    if vehicle.antique?
+      @collected_fees += 25
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+      vehicle.plate_type = :ev
+    else
+    @collected_fees += 100
+    vehicle.plate_type = :regular
+    end
+  end
```

Comment:
When a vehicle is registered, we also want to set a registration date. At the moment, in your vehicle class, your registration date is set upon initialization, but never updated once the vehicle is officially registered.